zipkin-storage/elasticsearch-http/src/main/java/zipkin/storage/elasticsearch/http/ElasticsearchHttpSpanConsumer.java

```diff
@@ -37,6 +38,15 @@
 
   final ElasticsearchHttpStorage es;
   final IndexNameFormatter indexNameFormatter;
+  private final static int MAX_CACHE_DAYS = 2;
+  private final static int MAX_CACHE_ENTRIES = 1000;
+
+  private final static Map<String, Set<Pair<String>>> indexToServiceSpansCache = Collections.synchronizedMap(new LinkedHashMap<String, Set<Pair<String>>>() {
```

Comment:
I wouldn't do synchronized map here, because you have a good bit of places which need atomic ops (put if absent). more straightforward to just synchronize when you need to imho